coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetViewHelperServiceImpl.java

```diff
@@ -269,4 +275,26 @@ private boolean isEmptyBudgetLineItemExists(Budget budget) {
         }
         return false;
     }
+
+    public void prepareHierarchySummary(ProposalBudgetForm form) {
+        if (form.getDevelopmentProposal().isInHierarchy()) {
+            form.setHierarchyDevelopmentProposals(getProposalHierarchyService().getHierarchyProposals(form.getDevelopmentProposal()));
+
+            for (DevelopmentProposal developmentProposal : form.getHierarchyDevelopmentProposals()) {
+                for (ProposalDevelopmentBudgetExt budget : developmentProposal.getBudgets()){
+                    if (budget.getBudgetSummaryDetails().isEmpty()){
+                        getBudgetCalculationService().populateBudgetSummaryTotals(budget);
+                    }
```

Comment:
Do we need to summarize all the budgets here? Can't we just do the last synced/final/newest?